changelog/12960.improvement.rst

```diff
@@ -0,0 +1 @@
+Removed _pytest.compat.is_generator() and just using inspect.isgeneratorfunction() directly
```

Comment:
This was a private function so we don't need to document it's removal, but we _do_ need to document the `breaking` change to yield tests, e.g.

> Test functions containing a `yield` now cause an explicit error.  They have not been run since Pytest 4.0, and were previously marked as an expected failure and deprecation warning.